MembraneRTC/src/main/java/org/membraneframework/rtc/media/SoundDetection.kt

```diff
@@ -0,0 +1,193 @@
+package org.membraneframework.rtc.media
+
+import android.annotation.SuppressLint
+import android.media.AudioFormat
+import android.media.AudioRecord
+import android.media.MediaRecorder
+import timber.log.Timber
+import java.util.*
+import kotlin.math.abs
+import kotlin.math.log10
+
+/**
+ * Class responsible for sound detection using the device's microphone.
+ */
+class SoundDetection {
+    private var timer: Timer? = null
+    private var audioRecord: AudioRecord? = null
+    private var onSoundDetectedListener: OnSoundDetectedListener? = null
+    private var bufferSize = 0
+    var isRecording = false
+        private set
+
+    /**
+     * Starts the sound detection process with the specified monitor interval, sampling rate, and volume threshold.
+     *
+     * @param monitorInterval The time interval (in milliseconds) between sound detection checks.
+     * @param samplingRate The audio sampling rate in Hz.
+     * @param volumeThreshold The threshold value in decibels (dB) above which a sound is considered detected.
+     */
+    @SuppressLint("MissingPermission")
+    fun start(monitorInterval: Int = 1, samplingRate: Int = 22050, volumeThreshold: Int = -100) {
+        if (isRecording) {
+            Timber.w("Sound detection is already in progress. Ignoring the start request.")
+            return
+        }
+
+        bufferSize = AudioRecord.getMinBufferSize(
+            samplingRate,
+            AudioFormat.CHANNEL_IN_MONO,
+            AudioFormat.ENCODING_PCM_16BIT
+        )
+        audioRecord = try {
+            AudioRecord(
+                MediaRecorder.AudioSource.MIC,
+                samplingRate,
+                AudioFormat.CHANNEL_IN_MONO,
+                AudioFormat.ENCODING_PCM_16BIT,
+                bufferSize
+            )
+        } catch (e: SecurityException) {
+            throw SecurityException(
+                "Unable to initialize the AudioRecord." +
+                    " Ensure that the recording permission is granted."
+            )
+        }
+
+        if (audioRecord?.state == AudioRecord.STATE_INITIALIZED) {
+            audioRecord?.startRecording()
+            isRecording = true
+            startTimer(monitorInterval, volumeThreshold)
+        } else {
+            Timber.e("COULDNT_PREPARE_RECORDING AudioRecord couldn't be initialized.")
+            throw RuntimeException("AudioRecord couldn't be initialized.")
```

Comment:
I'm not sure about throwing a `RuntimeException` here, maybe other type of exception would be more appropriate? See this discussion: https://stackoverflow.com/questions/2190161/difference-between-java-lang-runtimeexception-and-java-lang-exception

Especially if it can be that user "may be using a device that can't record audio." this is definitely a thing user might want to catch to show an alert or handle it somehow.